libnavigation-ui/src/main/java/com/mapbox/navigation/ui/VoiceInstructionCache.java

```diff
@@ -12,75 +12,73 @@
 
 class VoiceInstructionCache {
 
-  private static final int MAX_VOICE_INSTRUCTIONS_TO_CACHE = 10;
-  private static final int VOICE_INSTRUCTIONS_TO_CACHE_THRESHOLD = 5;
-  private final MapboxNavigation navigation;
-  private final VoiceInstructionLoader voiceInstructionLoader;
-  private final ConnectivityStatusProvider connectivityStatus;
-  private int totalVoiceInstructions = 0;
-  private int currentVoiceInstructionsCachedIndex = 0;
-  private boolean isVoiceInstructionsToCacheThresholdReached = false;
+    private static final int MAX_VOICE_INSTRUCTIONS_TO_CACHE = 10;
+    private static final int VOICE_INSTRUCTIONS_TO_CACHE_THRESHOLD = 5;
+    private final MapboxNavigation navigation;
+    private final VoiceInstructionLoader voiceInstructionLoader;
+    private final ConnectivityStatusProvider connectivityStatus;
+    private int totalVoiceInstructions = 0;
+    private int currentVoiceInstructionsCachedIndex = 0;
+    private boolean isVoiceInstructionsToCacheThresholdReached = false;
 
-  VoiceInstructionCache(MapboxNavigation navigation, VoiceInstructionLoader voiceInstructionLoader,
-                        ConnectivityStatusProvider connectivityStatus) {
-    this.navigation = navigation;
-    this.voiceInstructionLoader = voiceInstructionLoader;
-    this.connectivityStatus = connectivityStatus;
-  }
-
-  void preCache(DirectionsRoute route) {
-    if (!connectivityStatus.isConnected()) {
-      return;
+    VoiceInstructionCache(MapboxNavigation navigation, VoiceInstructionLoader voiceInstructionLoader,
+                          ConnectivityStatusProvider connectivityStatus) {
+        this.navigation = navigation;
+        this.voiceInstructionLoader = voiceInstructionLoader;
+        this.connectivityStatus = connectivityStatus;
     }
 
-    totalVoiceInstructions = 0;
-    currentVoiceInstructionsCachedIndex = 0;
-    isVoiceInstructionsToCacheThresholdReached = false;
-    for (int i = 0; i < route.legs().size(); i++) {
-      RouteLeg leg = route.legs().get(i);
-      for (int j = 0; j < leg.steps().size(); j++) {
-        LegStep step = leg.steps().get(j);
-        for (VoiceInstructions ignored : step.voiceInstructions()) {
-          totalVoiceInstructions++;
+    void preCache(DirectionsRoute route) {
+        if (!connectivityStatus.isConnected()) {
+            return;
         }
-      }
-    }
-    List<String> voiceInstructionsToCache = new ArrayList<>();
-    // TODO add to MapbxoNavigation retrieveSsmlAnnouncementInstruction from Navigator
```

Comment:
@RingerJK - the change in this class is un-comment the below code and the same one below.

Reply:
now it's obvious 🙇  thanks @JunDai !